### PR TITLE
Maint ad docs

### PR DESCRIPTION
**Pull request by vlipovac**
As discussed with @keileg , in order to establish the new workflow with regards to the docs branch, here an update to docstrings and type annotations of some AD classes and functions.
(src code synchronization with docs).

## Types of changes

- [ ] Minor change (e.g., dependency bumps, broken links, etc).
- [ ] Bugfix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Testing (contribution related to testing of existing or new functionality).
- [X] Documentation (contribution related to adding, improving, or fixing documentation).
- [X] Maintenance (e.g., improve logic and performance, remove obsolete code, etc).
- [ ] Other: 

## Checklist

_Put an `x` in the boxes that apply or explain briefly why the box is not relevant._

- [X] The documentation is up-to-date.	
- [X] Static typing is included in the update.
- [X] This PR does not duplicated existing functionality.
- [ ] The update is covered by the test suite (including tests added in the PR).

### CODE REVIEW

File: src/porepy/numerics/ad/operator_functions.py [hunk 1]
```diff
@@ -89,16 +88,16 @@ def __init__(
         self._operation: Operation = Operation.evaluate
         self._set_tree()
 
-    def __call__(self, *args: Union[pp.ad.Operator, Ad_array]):
+    def __call__(self, *args: pp.ad.Operator | Ad_array):
```
Comment (keileg) [inline src/porepy/numerics/ad/operator_functions.py:91]:
```suggestion
    def __call__(self, *args: pp.ad.Operator | Ad_array) -> pp.ad.Operator:
```
Is this correct?

**Review by keileg — COMMENTED** [review]
One minor comment. Besides: Thanks!

**Comment by vlipovac** [comment]
@keileg are the mypy issues still present?
The checks here fail in unchanged files.

**Comment by IvarStefansson** [comment]
> @keileg are the mypy issues still present? The checks here fail in unchanged files.

Checks pass on develop. So updating from develop should suffice.

**Comment by vlipovac** [comment]
> > @keileg are the mypy issues still present? The checks here fail in unchanged files.
> 
> Checks pass on develop. So updating from develop should suffice.

@IvarStefansson  it is up-to-date
image

**Comment by vlipovac** [comment]
> > @keileg are the mypy issues still present? The checks here fail in unchanged files.
> 
> Checks pass on develop. So updating from develop should suffice.

My bad, there was a return type annotation which was not in develop and which caused mistakes in the discretization file.

**Review by keileg — APPROVED** [review]

**Pull request merged**